### PR TITLE
Cleanup system includes

### PR DESCRIPTION
**Pull request by gunchleoc**
- Delete all system headers that the compiler won't ask for
- Deprecate `<ctype.h>`, `<io.h>` and `<stdint.h>` in favor of
  `<cctype>`, `<cstdio>` and `<cstdint>`
- Fix `USES_BOOST_REGEX` in CMakeLists.txt

### CODE REVIEW

**Comment by gunchleoc** [comment]
Travis is green, so this is ready or review now.

**Review by Noordfrees — APPROVED** [review]
Had a quick look at the diff – looks good. Compiles without new warnings or errors.
This can go in in my opinion as soon as appveyor is green.

File: cmake/codecheck/rules/deprecated_c_header.py [hunk 1]
```diff
@@ -36,11 +39,13 @@
 allowed = [
     '#include <cassert>',
     '#include <ccomplex>',
+    '#include <cctype>',
     '#include <ctype>',
     '#include <cerrno>',
     '#include <cfenv>',
     '#include <cfloat>',
     '#include <cinttypes>',
+    '#include <cstdio>',
```
Comment (rikardfalkeborn) [inline cmake/codecheck/rules/deprecated_c_header.py:48]: `<cstdio>` is already on the list (line 59).

Comment (gunchleoc) [reply cmake/codecheck/rules/deprecated_c_header.py:48]: Spam?

**Comment by gunchleoc** [comment]
AppVeyor failure is just the usual timeout, so I'm gonna merge. Thanks for the review!

**Pull request merged**